### PR TITLE
Expand the stack to include client-side frameworks

### PR DESCRIPTION
**Pull request by jferris**
This documents the conclusions from our mobile and UI strategy sessions.
It also splits out native iOS and Android from the Core Stack,
maintaining them as specialties.

### CODE REVIEW

File: stack/README.md [hunk 1]
```diff
@@ -15,50 +15,28 @@ inappropriate for the task at hand. This allows us to practice our value of
...
-We recommend learning at least one of the following:
-
-* MapKit or Google Maps
-* Core Location
-* Core Bluetooth
-* PhotoKit
-* UserNotifications
+* Use React when building components with a client-side framework.
+* Use TypeScript when writing client-side code.
+* Avoid building single-page applications for the web.
```
Comment (JoelQ) [inline stack/README.md:32]: 💯

File: stack/README.md [hunk 2]
```diff
@@ -15,50 +15,28 @@ inappropriate for the task at hand. This allows us to practice our value of
...
-* Prefer standard UIKit components to custom views.
-
-We recommend learning at least one of the following:
-
-* MapKit or Google Maps
-* Core Location
-* Core Bluetooth
-* PhotoKit
-* UserNotifications
+* Use React when building components with a client-side framework.
```
Comment (JoelQ) [inline stack/README.md:30]: Much as I like Elm, I agree that React + Typescript makes more sense as our default.

File: stack/README.md [hunk 3]
```diff
@@ -15,50 +15,28 @@ inappropriate for the task at hand. This allows us to practice our value of
...
+* Use TypeScript when writing client-side code.
+* Avoid building single-page applications for the web.
+* When building a cross-platform mobile app that will be delivered via an app
+  store, use React Native.
 
 ### API
 
 * Use GraphQL as an API layer when connecting a mobile app to a web service.
+* Avoid using GraphQL when connecting from one service to another.
```
Comment (JoelQ) [inline stack/README.md:39]: Hmmm I'm not sure what this means? Do we prefer GraphQL for APIs used by frontends but REST for APIs used by other backends?

Comment (mike-burns) [reply stack/README.md:39]: +1 to removing this, or replacing it with a "Prefer" statement that we can bikeshed instead.

Comment (jferris) [reply stack/README.md]: I removed it in a663ab7.

**Comment by WilHall** [comment]
@jferris Thoughts on merging this once the `README.md` conflicts are resolved?

File: stack/README.md [hunk 4]
```diff
@@ -15,46 +15,23 @@ inappropriate for the task at hand. This allows us to practice our value of
...
-allows us to swap out an entire layer when necessary without losing all the
-decisions made for other layers.
+Most developers at thoughtbot learn our Core Stack. The stack is broken up into
+layers and each layer depends on another layer. This allows us to swap out an
+entire layer when necessary without losing all the decisions made for other
+layers. Developers will have varying levels of experience with each layer, but
+the gaps between layers are small enough that developers can learn a new layer
+while building applications.
 
 ### UI
```
Comment (mike-burns) [inline stack/README.md:27]: Should this be "Web UI" instead? I don't expect e.g. Android UI to use server-rendered HTML.

Comment (jferris) [reply stack/README.md:27]: This is also attempting to say that an HTML-based web UI will be the default UI. If we don't need to build a mobile application, we won't.

This section and the server-rendered HTML guideline weren't added by this pull request, so I would open a new issue or pull request if you think it should change.

**Comment by iwz** [comment]
This is nice. I'm curious about the "components" part. React is very clearly component-based. Is this guideline to differentiate between choosing React and web components? 

What about if not building with components? Where do other tools like Stimulus and Vue fit in?

**Pull request merged**